lib/matplotlib/axes/_axes.py

```diff
@@ -6211,6 +6218,10 @@ def pcolorfast(self, *args, alpha=None, norm=None, cmap=None, vmin=None,
                     else:
                         style = "pcolorimage"
             elif x.ndim == 2 and y.ndim == 2:
+                if C.ndim > 2:
+                    raise TypeError('''
```

Comment:
Yes ValueError.  This message doesn't actually make much sense to me; I don't think a naive user has any idea what Quadmesh is, nor will they care.  Can this error message just say what the error is and how to fix it without asking any questions...

Reply:
I've changed the error type and tried to write a better error message